### ISSUE TITLE
translation fails when literal block is in a list 

### ISSUE THREAD

**Issue by nobodxbodon** [sphinx-doc/sphinx#4913]
Excuse me if this is a user issue, as I'm new to reStructuredText and Sphinx.
[After putting the literal block into a list](https://github.com/nobodxbodon/sphinx/commit/73bcc09fa8a69f2ca6da504fdc1d0eb0c0231cca#diff-b93bc6f14f5b7f025297fb647fbcaeaeR23), this test case fails:
`$ tox -e py36 tests/test_intl.py::test_additional_targets_should_be_translated`

**Comment by tk0miya** [comment]
It seems docutils (reST parser) does not assign source filename to literal_block under bullet list:
```
from docutils import nodes
from docutils.core import publish_doctree


text = ("::\n"
        "\n"
        "  literal_block on root\n"
        "\n"
        "* ::\n"
        "\n"
        "    literal_block inside bullet_list\n")
doctree = publish_doctree(text, source_path='<string>')

# literal_block on root has correct source
assert isinstance(doctree[0], nodes.literal_block)
assert doctree[0].source == '<string>'
assert doctree[0].line == 3

# bullet_list also has correct source
assert isinstance(doctree[1], nodes.bullet_list)
assert doctree[1].source == '<string>'
assert doctree[1].line == 5

# but literal_block inside bullet_list does not
assert isinstance(doctree[1][0][0], nodes.literal_block)
assert doctree[1][0][0].source == '<string>'
assert doctree[1][0][0].line == 7
```
```
$ python test_literal_block.py
Traceback (most recent call last):
  File "test_literal_block.py", line 26, in <module>
    assert doctree[1][0][0].source == '<string>'
AssertionError
```

**Comment by tk0miya** [comment]
Fixed by #4943
